src/window.rs

```diff
@@ -18,6 +18,27 @@ use crate::{
 
 pub use crate::icon::{BadIcon, Icon};
 
+/// Progress State
+pub enum ProgressState {
+  None,
+  Normal,
+  /// **Treated as Normal in linux**
+  Intermediate,
+  /// **Treated as Normal in linux**
+  Paused,
+  /// **Treated as Normal in linux**
+  Error,
+}
+
+pub struct ProgressBarState {
+  /// The progress bar state.
+  pub state: Option<ProgressState>,
+  /// The progress bar progress. This can be a value ranging from `0` to `100`
```

Comment:
How about using `f64` and ranging the progress from 0 to 1?

Since most use cases are only a part of something, we don't need to multiply by 100 every time we calculate the progress.

Reply:
Then too you'll need to multiply it by 100, because the windows impl takes u64

Reply:
Multiply the value first is a bit unnecessary because each platform takes different input like:
- Linux takes a `double` value from `0.0` to `1.0`
- Windows takes `0` to `100`
- macOS I think it defaults to `0` to `100` too, but it can be set to `0.0` to `1.0` as well

So either way we need to do the math on the `platform_impl` side. But yes, letting the range from `0` to `100` is much more intuitive when you want to set progress directly.

Reply:
There are two platforms using integer. I think it's better to stick to 0-100.
float causes more operation overhead than multiply imho.

Reply:
Yeah totally agree, computers suck at decimals